src/population/simple_population.py

```diff
@@ -77,6 +83,8 @@ def integrate_generation(self, genomes: List[Optional[G]]) -> None:
             key=lambda g: g.fitness, reverse=True
         )
 
+        self.track_all_genomes()
```

Comment:
I would recommend making your `FamilyTreeTracker` a child class of `LogDataProvider[Population]` and calling this in the `get_log_data` method. So you will end up with something like this:


```
class FamilyTreeTracker(LogDataProvider[Population[EXAStarGenome]]):


    @overrides(LogDataProvider)
    def get_log_data(self, aggregator: Population[EXAStarGenome]) -> Dict[str, Any]:
        """
        Args:
            aggregator: A reference to the aggregator this LogDataProvider is associated with (see below).
              This value may be necessary to obtain the data you want to log.

        Returns:
            A dictionary of log data. The keys in this dictionary should be unique.
        """
        # I just realized that we probably need to add a Population::get_genomes method.
        self.track_genomes(aggregator.get_genomes())
        return {} # we don't actually have any data for the main log, since this class creates its own.
```

We will also need to add a method to `Population` something like:
````
    def get_genomes(self) -> Sequence[G]:
        """
        Return:
            A sequence of all genomes currently contained in this population. Modification of this sequence is permitted, so it should be a copy of any underlying sequence.
        """
````